core/src/main/java/org/mule/runtime/core/internal/exception/ErrorHandler.java

```diff
@@ -78,9 +78,10 @@ public Event handleException(MessagingException exception, Event event) {
       return error(exception);
     } else {
       Event event = addExceptionPayload(exception, exception.getEvent());
+      exception.setProcessedEvent(event);
       for (MessagingExceptionHandlerAcceptor exceptionListener : exceptionListeners) {
         if (exceptionListener.accept(event)) {
-          return exceptionListener.apply(new MessagingException(event, exception));
+          return exceptionListener.apply(exception);
```

Comment:
is this change needed? we should aim to make `MessagingException` immutable eventually